binary.json

```diff
@@ -46,8 +46,13 @@
          "expected": 31
       },
       {
-         "description": "invalid binary numbers raise an error",
-         "binary": ["012", "10nope", "nope10", "10nope10", "001 nope", "2"],
+         "description": "numbers other than one and zero raise an error",
+         "binary": ["012", "2"],
```

Comment:
This should become 2 tests, 
```
         "description": "2 is not a valid binary digit and so should raise an error",
         "binary":  "2",
         "expected" :  null,
```
and
```
         "description": "A number containing an invalid binary digit is invalid and should raise an error",
         "binary":  "01201",
         "expected":  null,
```

Discussion required: I'm not 100% sure null is the correct expected value here if we want to indicate in a general way that an error should be raised.